packages/demo-cards/claim.gts

```diff
@@ -44,17 +56,27 @@ class Isolated extends Component<typeof Claim> {
   constructor(owner: unknown, args: any) {
     super(owner, args);
     this.initialize.perform();
+    if (this.isMetamaskInstalled) {
+      window.ethereum.on('chainChanged', (chainId: string) => {
+        this.connected =
+          parseInt(chainId, 16) == this.args.model.chain?.chainId;
+        window.location.reload(); // metamask recommends to reload page
+      });
+    }
   }
-  private initialize = restartableTask(async () => {
-    let metamaskChainId = await this.getChainId();
-    let isChainEqual = this.args.model.chain?.chainId == metamaskChainId;
+  private initialize = enqueueTask(async () => {
```

Comment:
I had a lot of issues when randomly swapping routes Claim/1 and Claim/2. It turns out I had to use `enueue` becos the `restartable` wud be cancelled. This is robust. `restartable` should be used for the buttons we click